composer.json

```diff
@@ -19,7 +19,7 @@
         "source": "https://github.com/colinodell/psr-testlogger"
     },
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^8.0",
```

Comment:
According to `semver` this means it should be version `2.0.0`? 🤔

Reply:
semver doesn't tie you to a particular PHP version support cycle, it's up to the project to determine what counts as a breaking API change and document it, and if PHP version increases don't count, then they don't.

Reply:
Correct, a bump in dependencies is not a BC break - see https://github.com/semver/semver/blob/d460299638d3d2af6fcf809285dc6a469e15742a/semver.md#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api

The only reason we might need to release this as a new major version would be if we BC in the public API. Although the API signature of `TestLogger` _did_ change, all of those changes are backward-compatible:

 - The class is `final`, so we don't need to worry about child classes suddenly becoming incompatible
 - The only type changes were parameter type widening. The new `string|int` types will continue to accept `string` inputs with no changes needed by end users.

> I'm fairly sure this would make the Drupal maintainers feel uneasy because if this happens again at some point in the future, then it technically breaks backwards compatibility for Drupal

This shouldn't be an issue as Composer will prevent the installation of newer versions with incompatible dependencies.

For example, this change will be released as v1.2.0, so I imagine Drupal will use a constraint like `^1.2`.  If I were to release v1.3.0 in the future with a new minimum PHP version of `8.2.0`, Composer would still allow v1.2.0 on PHP 8.0 and 8.1 - only users on 8.2 could install v1.3.0.  And because it would be a minor version bump and we follow semver, there would be no breaking changes to the API.  This means that both versions would function identically as far as Drupal is concerned.

Bumping dependencies (including minimum PHP versions) in a minor release is done by many other large PHP projects, including Doctrine: https://www.doctrine-project.org/2017/07/25/php-7.1-requirement-and-composer.html

**TL;DR:** Everything will "just work" :)